### ISSUE TITLE
Add ability to run commands in a scheduled context

### ISSUE THREAD

**Issue by dubzzz** [dubzzz/fast-check#482]
Follow up of https://github.com/dubzzz/fast-check/pull/479

With the upcoming addition of `fc.schedule`, one common need will be to execute a sequence of operations while interleaving resolved promises in-between those ones.

For instance, if I want to test an autocomplete field fetching data on the network I can come with the following scenario (to check potential race conditions):

- Enter some letters corresponding to the user's query
- Wait for some queries to be resolved (maybe not all, maybe not in the right order, maybe zero)
- Enter new letters

And each time, I'll check that both the suggestions displayed on my user's screen and the content of the autocomplete match.

The example is easy to build with suggested implementation of `fc.schedule` - see https://github.com/dubzzz/fast-check/pull/479

However, ones might not just to type characters but also to click on the 'clear field' button or any other capacity offered by our component. This kind of capability is already offered by our model based testing system but unfortunately it does not deal with interleaving promises in-between our commands.

The aim of this issue is to add this capability

**Comment by hamiltop** [comment]
I'm trying to flip this around a bit.

As I understand it, this is about looking for race conditions in using the results of an async action. It does not enable testing of side-effects of async actions.

A simple example of what I'd like to do is to call an API that has a race condition. The code in the API might look like:
```
const count = await getCountFromDB();
await updateCountInDB(count + 1);
```
Clearly, concurrent calls to this API are going to interfere and result in an incorrect count.

Is there a way to use the scheduler to accomplish this test?

**Comment by dubzzz** [comment]
@hamiltop 

You're perfectly right, the idea is to provide a built-in way to test race conditions directly with fast-check. Behind the hood, fast-check will use its property based capabilities to re-order when promises will be resolved.

The case with `getCountFromDB` and `updateCountInDB` is already possible with the latest release of fast-check. I just opened a PR to add this example in the list of available examples I have in the doc: https://github.com/dubzzz/fast-check/pull/554/files#diff-72bba4c2c440aafce4e4fc82d55f5fb9

---

The current PR tries to merge race condition detection capabilities of fast-check wth its model based testing approach (about model based testing: [documented here](https://github.com/dubzzz/fast-check/blob/master/documentation/1-Guides/Tips.md#model-based-testing-or-ui-test) and [explained here](https://medium.com/criteo-labs/detecting-the-unexpected-in-web-ui-fuzzing-1f3822c8a3a5)).

In a nutshell, in classical model based approach the commands were executed sequentially. In the proposed approach, commands are still executed sequentially but they can be interleaved by the API calls they triggered.

For instance, in the example https://github.com/dubzzz/fast-check/pull/491/files#diff-e6717ef357ebadda92f37eedb9004651, my commands are: add, toggle, remove. Whenever I call one of those commands, they may start asynchronous calls to an API. With this new approach API calls can also be scheduled so they may be resolved later or immediately after the command.

---

Please let me know what you think about the test added in #554 

Having an easy API for race conditions is not so easy so the more feedbacks I have the better I can work on examples to better explain it
